mne/io/eeglab/eeglab.py

```diff
@@ -0,0 +1,410 @@
+    return epochs
+
+
+class RawEEGLAB(_BaseRaw):
+    """Raw object from EEGLAB .set file.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to the .set file. If the data is stored in a separate .fdt file,
+        it is expected to be in the same folder as the .set file.
+    montage : str | None | instance of montage
+        Path or instance of montage containing electrode positions.
+        If None, sensor locations are (0,0,0). See the documentation of
+        :func:`mne.channels.read_montage` for more information.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of RawEEGLAB
+        A Raw object containing EEGLAB .set data.
+
+    Notes
+    -----
+    .. versionadded:: 0.11.0
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    @verbose
+    def __init__(self, input_fname, montage, preload=False, verbose=None):
+        """Read EEGLAB .set file.
+        """
+        from scipy import io
+        basedir = op.dirname(input_fname)
+        _check_mat_struct(input_fname)
+        eeg = io.loadmat(input_fname, struct_as_record=False,
+                         squeeze_me=True)['EEG']
+        if eeg.trials != 1:
+            raise TypeError('The number of trials is %d. It must be 1 for raw'
+                            ' files. Please use `mne.io.read_epochs_eeglab` if'
+                            ' the .set file contains epochs.' % eeg.trials)
+
+        last_samps = [eeg.pnts - 1]
+        info = _get_info(eeg, montage)
+        # read the data
+        if isinstance(eeg.data, string_types):
+            data_fname = op.join(basedir, eeg.data)
+            _check_fname(data_fname)
+            logger.info('Reading %s' % data_fname)
+
+            super(RawEEGLAB, self).__init__(
+                info, preload, filenames=[data_fname], last_samps=last_samps,
+                orig_format='double', verbose=verbose)
+        else:
+            if preload is False or isinstance(preload, string_types):
+                warnings.warn('Data will be preloaded. preload=False or a'
+                              ' string preload is not supported when the data'
+                              ' is stored in the .set file')
+            data = eeg.data.reshape(eeg.nbchan, -1, order='F')
+            data = data.astype(np.double)
+            super(RawEEGLAB, self).__init__(
+                info, data, last_samps=last_samps, orig_format='double',
+                verbose=verbose)
```

Comment:
The alternative approach is to do:
```
if preload is False ...:
    ...
super(RawEEGLAB, self).__init__(info, True, ...)
```
In other words, get rid of your `data = ` lines and let the super class do the data preloading itself. This is better because the user will get `logger.info` messages that you will not give them here, and it is more DRY because it will use the nice reading function you already made.